### ISSUE TITLE
error: could not compile `wabt-sys` due to previous error

### ISSUE THREAD

**Issue by junkei-okinawa** [demergent-labs/kybra#30]
hello.
It's a great project and I've been looking forward to it being available in alpha.

When I tried to deploy, I got the following error.

I'm using Apple Silicon (M1 chip).
```zsh
$dfx -V       
dfx 0.12.0-beta.2

$cargo -V                        
cargo 1.64.0 (387270bc7 2022-09-16)

$rustc -V     
rustc 1.64.0 (a55dd71d5 2022-09-19)

$python -V        
Python 3.9.7
```

```zsh
Deploying all canisters.
Creating canisters...
Creating canister imports...
imports canister created with canister id: rkp4c-7iaaa-aaaaa-aaaca-cai
Building canisters...
Executing 'python -m kybra imports src/main.py src/main.did'
info: component 'rust-std' for target 'wasm32-unknown-unknown' is up to date
    Updating crates.io index
  Installing ic-cdk-optimizer v0.3.4
   Compiling proc-macro2 v1.0.44
   Compiling quote v1.0.21
   Compiling unicode-ident v1.0.4
   Compiling cc v1.0.73
   Compiling memchr v2.5.0
   Compiling version_check v0.9.4
   Compiling syn v1.0.101
   Compiling unicode-segmentation v1.10.0
   Compiling regex-syntax v0.6.27
   Compiling serde v1.0.145
   Compiling autocfg v1.1.0
   Compiling libc v0.2.133
   Compiling serde_json v1.0.85
   Compiling serde_derive v1.0.145
   Compiling hashbrown v0.12.3
   Compiling ryu v1.0.11
   Compiling itoa v1.0.3
   Compiling unicode-width v0.1.10
   Compiling os_str_bytes v2.4.0
   Compiling lazy_static v1.4.0
   Compiling termcolor v1.1.3
   Compiling bitflags v1.3.2
   Compiling strsim v0.10.0
   Compiling vec_map v0.8.2
   Compiling humansize v1.1.1
   Compiling heck v0.3.3
   Compiling cmake v0.1.48
   Compiling proc-macro-error-attr v1.0.4
   Compiling proc-macro-error v1.0.4
   Compiling textwrap v0.12.1
   Compiling indexmap v1.9.1
   Compiling wabt-sys v0.8.0
   Compiling aho-corasick v0.7.19
   Compiling regex v1.6.0
   Compiling atty v0.2.14
   Compiling binaryen-sys v0.12.0
   Compiling clap_derive v3.0.0-beta.2
   Compiling clap v3.0.0-beta.2
   Compiling wabt v0.10.0
error: failed to add native library /var/folders/0h/1jsn6jwn7sz_s6zm5tk2znkw0000gn/T/cargo-installAGJQUZ/release/build/wabt-sys-21af77b6012a4d59/out/build/libwabt.a: file too small to be an archive

error: could not compile `wabt-sys` due to previous error
```

Looking for a solution, I ran cargo install ic-cdk-optimizer --version 0.3.4 directly in terminal and removed it from build.sh and it worked.

```build.sh
rustup target add wasm32-unknown-unknown
# cargo install ic-cdk-optimizer --version 0.3.4

```

**Comment by lastmjs** [comment]
Looks like the same issue as here: https://github.com/demergent-labs/azle/issues/237

**Comment by lastmjs** [comment]
I will try to debug this on an M1 and get a fix in, we might also be able to remove the need for ic-cdk-optimizer and use ic-wasm instead.

**Comment by lastmjs** [comment]
It looks like we just need to follow the pattern here: https://github.com/demergent-labs/azle/blob/main/src/azle.ts#L138

I should have this done soon

**Comment by lastmjs** [comment]
For now we'll put in the documentation that Mac M1 users have to manually install ic-cdk-optimizer, but that should be it

**Comment by lastmjs** [comment]
I'm hoping this gets resolved once we switch to ic-wasm: https://github.com/demergent-labs/kybra/issues/42

**Comment by lastmjs** [comment]
This should be resolved as of Kybra 0.4.0
